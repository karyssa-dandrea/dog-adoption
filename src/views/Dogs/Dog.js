import React from 'react';
import { fetchDogById } from '../../services/dogs';
import { useState, useEffect } from 'react';
import DogDetail from '../../components/DogDeets/DogDetails';

export default function Dog(props) {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h3>Loading</h3>;
  }

  return (
    <div className="dog-deets">
      <DogDetail dog={dog} />
    </div>
  );
}
