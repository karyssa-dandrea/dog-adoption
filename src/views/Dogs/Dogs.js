import React from 'react';
import { fetchDogs } from '../../services/dogs';
import { useState, useEffect } from 'react';
import DogList from '../../components/DogDeets/DogList';
import Header from '../../components/Header/Header';

export default function Dogs() {
  const [dogs, setDogs] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogs();
      setDogs(data);
      setLoading(false);
    };
    fetchData();
  }, []);

  if (loading) {
    return <h3>Loading</h3>;
  }

  return (
    <div>
      <Header />
      <DogList dogs={dogs} />
    </div>
  );
}
