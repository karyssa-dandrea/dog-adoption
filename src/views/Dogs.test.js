import { render, screen } from '@testing-library/react';
import Dogs from '../views/Dogs/Dogs';
import { MemoryRouter } from 'react-router-dom';

test('should render list of dogs', async () => {
  const { container } = render(
    <MemoryRouter>
      <Dogs />
    </MemoryRouter>
  );
  await screen.findByText('Meet Benson');
  expect(container).toMatchSnapshot();
});
