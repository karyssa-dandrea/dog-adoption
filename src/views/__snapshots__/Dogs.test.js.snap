// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render list of dogs 1`] = `
<div>
  <div>
    <div>
      <header>
        <a
          href="/dogs"
        >
           Home 
        </a>
      </header>
    </div>
    <div
      class="container"
    >
      <a
        class="lists"
        href="/dogs/3"
      >
        <h1>
          Meet 
          Benson
        </h1>
        <img
          src="https://placedog.net/samples/205.jpg"
        />
        <p>
          Benson
           is a(n) 
           year old 
          Australian Shepard
        </p>
      </a>
      <a
        class="lists"
        href="/dogs/6"
      >
        <h1>
          Meet 
          Ginny
        </h1>
        <img
          src="https://placedog.net/samples/188.jpg"
        />
        <p>
          Ginny
           is a(n) 
           year old 
          Border Collie
        </p>
      </a>
      <a
        class="lists"
        href="/dogs/4"
      >
        <h1>
          Meet 
          Maggie
        </h1>
        <img
          src="https://placedog.net/samples/178.jpg"
        />
        <p>
          Maggie
           is a(n) 
           year old 
          Australian Shepard
        </p>
      </a>
      <a
        class="lists"
        href="/dogs/1"
      >
        <h1>
          Meet 
          Nori
        </h1>
        <img
          src="https://placedog.net/samples/197.jpg"
        />
        <p>
          Nori
           is a(n) 
           year old 
          Labrador Retriever 
        </p>
      </a>
      <a
        class="lists"
        href="/dogs/2"
      >
        <h1>
          Meet 
          Tokio
        </h1>
        <img
          src="https://placedog.net/samples/200.jpg"
        />
        <p>
          Tokio
           is a(n) 
           year old 
          Labradoodle
        </p>
      </a>
      <a
        class="lists"
        href="/dogs/5"
      >
        <h1>
          Meet 
          Watson
        </h1>
        <img
          src="https://placedog.net/samples/158.jpg"
        />
        <p>
          Watson
           is a(n) 
           year old 
          Labrador Retriever 
        </p>
      </a>
    </div>
  </div>
</div>
`;
